tests/metal/simple-task.slang

```diff
@@ -0,0 +1,102 @@
+//TEST:SIMPLE(filecheck=CHECK): -target metal
+//TEST:SIMPLE(filecheck=CHECK-ASM): -target metallib
+//TEST:REFLECTION(filecheck=REFLECT):-target metal -entry main_kernel -stage task
+
+// TEST_INPUT: ubuffer(data=[0 0 0 0], stride=4):out,name outputBuffer
+
+uniform RWStructuredBuffer<float> outputBuffer;
+
+cbuffer Uniforms
+{
+    float4x4 modelViewProjection;
+}
+
+// CHECK-ASM: define void @taskMain
```

Comment:
There is no //CHECK: lines in the file, so test in line 1 will fail

Reply:
I'm not entirely sure what those check comments do, copied them from the rasterization test file.

Reply:
They are regular expressions used to match the result of the generated source file or buffer content.

A //CHECK line will try to match a line in the output with the regex following `CHECK:`, if such a match if found, it is successful, otherwise it will fail the test.